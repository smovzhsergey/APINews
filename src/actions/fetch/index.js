// Core
import types from './types';

export default Object.freeze({
    fetchSources: () => ({
        type: types.FETCH_SOURCES
    }),
    fetchSourcesSuccess: (sources) => ({
        type:    types.FETCH_SOURCES_SUCCESS,
        payload: sources
    }),
    fetchSourcesFail: (message) => ({
        type:    types.FETCH_SOURCES_FAIL,
        payload: message
    }),
    fetchNews: () => ({
        type: types.FETCH_NEWS
    }),
    fetchNewsSuccess: (news) => ({
        type:    types.FETCH_NEWS_SUCCESS,
        payload: news
    }),
    fetchNewsFail: (message) => ({
        type:    types.FETCH_NEWS_FAIL,
        payload: message
    }),
    fillDefaultSources: (sources) => ({
        type:    types.FILL_DEFAULT_SOURCES,
        payload: sources
    }),
    startFetching: () => ({
        type: types.START_FETCHING
    }),
    stopFetching: () => ({
        type: types.STOP_FETCHING
    })
});
