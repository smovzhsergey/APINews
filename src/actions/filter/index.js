// Core
import types from './types';

export default Object.freeze({
    getSelectedSources: (sources) => ({
        type:    types.GET_SELECTED_SOURCES,
        payload: sources
    }),
    getSelectedCategory: (category) => ({
        type:    types.GET_SELECTED_CATEGORY,
        payload: category
    }),
    resetFilter: () => ({
        type: types.RESET_FILTER
    })
});
