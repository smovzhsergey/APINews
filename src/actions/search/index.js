// Core
import types from './types';

export default Object.freeze({
    clearSearchValue: () => ({
        type: types.CLEAR_SEARCH_VALUE
    }),
    changeSearchValue: (value) => ({
        type:    types.CHANGE_SEARCH_VALUE,
        payload: value
    })
});
