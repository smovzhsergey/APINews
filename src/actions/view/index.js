// Core
import types from './types';

export default Object.freeze({
    openFilterPanel: () => ({
        type: types.OPEN_FILTER_PANEL
    }),
    closeFilterPanel: () => ({
        type: types.CLOSE_FILTER_PANEL
    }),
    showButtonUp: () => ({
        type: types.SHOW_BUTTON_UP
    }),
    hideButtonUp: () => ({
        type: types.HIDE_BUTTON_UP
    }),
    addNewsInView: () => ({
        type: types.ADD_NEWS_IN_VIEW
    }),
    resetNewsInView: () => ({
        type: types.RESET_NEWS_IN_VIEW
    }),
    updateNews: () => ({
        type: types.UPDATE_NEWS
    }),
    clearNewsUpdate: () => ({
        type: types.CLEAR_NEWS_UPDATE
    }),
    setQuantityNewNews: (quantity) => ({
        type:    types.SET_QUANTITY_NEW_NEWS,
        payload: quantity
    })
});
