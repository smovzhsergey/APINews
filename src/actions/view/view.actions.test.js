//Core
import actions from './';

const quantity = 5;

describe('view actions: ', () => {
    test('"openFilterPanel" actions creator should produce a corresponding action', () => {
        expect(actions.openFilterPanel()).toMatchSnapshot();
    });
    test('"closeFilterPanel" actions creator should produce a corresponding action', () => {
        expect(actions.closeFilterPanel()).toMatchSnapshot();
    });
    test('"showButtonUp" actions creator should produce a corresponding action', () => {
        expect(actions.showButtonUp()).toMatchSnapshot();
    });
    test('"hideButtonUp" actions creator should produce a corresponding action', () => {
        expect(actions.hideButtonUp()).toMatchSnapshot();
    });
    test('"addNewsInView" actions creator should produce a corresponding action', () => {
        expect(actions.addNewsInView()).toMatchSnapshot();
    });
    test('"resetNewsInView" actions creator should produce a corresponding action', () => {
        expect(actions.resetNewsInView()).toMatchSnapshot();
    });
    test('"updateNews" actions creator should produce a corresponding action', () => {
        expect(actions.updateNews()).toMatchSnapshot();
    });
    test('"clearNewsUpdate" actions creator should produce a corresponding action', () => {
        expect(actions.clearNewsUpdate()).toMatchSnapshot();
    });
    test('"setQuantityNewNews" actions creator should produce a corresponding action', () => {
        expect(actions.setQuantityNewNews(quantity)).toMatchSnapshot();
    });
});
