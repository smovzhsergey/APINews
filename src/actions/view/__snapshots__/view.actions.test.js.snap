// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`view actions:  "addNewsInView" actions creator should produce a corresponding action 1`] = `
Object {
  "type": "ADD_NEWS_IN_VIEW",
}
`;

exports[`view actions:  "clearNewsUpdate" actions creator should produce a corresponding action 1`] = `
Object {
  "type": "CLEAR_NEWS_UPDATE",
}
`;

exports[`view actions:  "closeFilterPanel" actions creator should produce a corresponding action 1`] = `
Object {
  "type": "CLOSE_FILTER_PANEL",
}
`;

exports[`view actions:  "hideButtonUp" actions creator should produce a corresponding action 1`] = `
Object {
  "type": "HIDE_BUTTON_UP",
}
`;

exports[`view actions:  "openFilterPanel" actions creator should produce a corresponding action 1`] = `
Object {
  "type": "OPEN_FILTER_PANEL",
}
`;

exports[`view actions:  "resetNewsInView" actions creator should produce a corresponding action 1`] = `
Object {
  "type": "RESET_NEWS_IN_VIEW",
}
`;

exports[`view actions:  "setQuantityNewNews" actions creator should produce a corresponding action 1`] = `
Object {
  "payload": 5,
  "type": "SET_QUANTITY_NEW_NEWS",
}
`;

exports[`view actions:  "showButtonUp" actions creator should produce a corresponding action 1`] = `
Object {
  "type": "SHOW_BUTTON_UP",
}
`;

exports[`view actions:  "updateNews" actions creator should produce a corresponding action 1`] = `
Object {
  "type": "UPDATE_NEWS",
}
`;
